android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -296,32 +296,33 @@ public void onListItemClick(ListView l, View v, int position, long id) {
 
   @Override
   public void onUrlDeviceDiscoveryUpdate() {
+    Log.v(TAG, "onUrlDeviceDiscoveryUpdate");
+    if (SwipeDismissListViewTouchListener.isLocked()) {
+      return;
+    }
+    for (PwPair pwPair : mPwCollection.getGroupedPwPairsSortedByRank(
+        new Utils.PwPairRelevanceComparator())) {
+      String groupId = Utils.getGroupId(pwPair.getPwsResult());
+      Log.d(TAG, "groupid to add " + groupId);
+      if (mNearbyDeviceAdapter.containsGroupId(groupId)) {
+        mNearbyDeviceAdapter.updateItem(pwPair);
+      } else if (!mGroupIdQueue.contains(groupId)
+          && !Utils.isBlocked(pwPair)) {
+        mGroupIdQueue.add(groupId);
+      }
+    }
+    // If queue is empty or the second scan is not complete, do not
+    // try to empty queue. Causes blocky UI behavior
```

Comment:
Remove the comment about blocky UI behavior...that's not the core reason to check, and it shouldn't inherently cause janky behavior.